src/monodroid/jni/monodroid-glue.c

```diff
@@ -535,6 +536,28 @@ get_xamarin_android_msbuild_path (void)
 	return msbuild_folder_path;
 }
 
+static char *libmonoandroid_directory_path = NULL;
+
+// Returns the directory in which this library was loaded from
+static char*
+get_libmonoandroid_directory_path ()
+{
+	wchar_t module_path[MAX_PATH];
+	char *converted_path = NULL;
+	HMODULE module = NULL;
+
+	if (libmonoandroid_directory_path != NULL)
+		return libmonoandroid_directory_path;
+
+	if (!GetModuleHandleEx (GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+	                        (LPCSTR)&libmonoandroid_directory_path, &module))
+		return NULL;
+	GetModuleFileNameW (module, module_path, sizeof (module_path));
```

Comment:
`sizeof (module_path)` is wrong here. [According to the `GetModuleFileName()` docs](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683197(v=vs.85).aspx):

> *nSize* [in]  
> > The size of the *lpFilename* buffer, in **TCHARs**.

Alas, this is *not* `sizeof(module_path)`. This is instead `sizeof(module_path)/sizeof(module_path[0])`.

Aside: this is the *first* time I've ever even heard of the `_pgmptr` global variable, encountered from reading those docs...